### PR TITLE
Resolves #432: Add new GET firewall endpoint and associated model

### LINKED ISSUE #432: Add new GET firewall endpoint and associated model
We have a new endpoint to add for retrieving an edge firewall service. (this new endpoint will supersede the previous one)

### PR DESCRIPTION
**Pull request by fradinni**

### CODE REVIEW

File: src/sdk/model/edge/edge.ts [hunk 1]
```diff
@@ -218,32 +220,45 @@ export class Edge extends Entity {
 
   /**
    * Get the Edge firewall
+   * @deprecated This method is deprecated, use getEdgeFirewall() instead
    * @returns {Promise<Firewall>}
    */
-  async getFirewall(): Promise<Firewall> {
+  async getFirewall(): Promise<EdgeFirewall> {
```
Comment (cfsnyder) [inline src/sdk/model/edge/edge.ts]: @fradinni so `getFirewall` returns `EdgeFirewall` and `getEdgeFirewall` returns `Firewall`? Doesn't that seem backwards?

File: src/sdk/model/edge/firewall/deprecated/edge-firewall-rule.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
+import { EdgeFirewallRuleJson } from './__json__/edge-firewall-rule-json';
+
+/**
+ * FirewallRule class
+ */
+export class EdgeFirewallRule {
```
Comment (cfsnyder) [inline src/sdk/model/edge/firewall/deprecated/edge-firewall-rule.ts]: did you generate these with the IDEA plugin or write them manually?

Comment (fradinni) [reply src/sdk/model/edge/firewall/deprecated/edge-firewall-rule.ts]: This class is the old one ( everything that is in /deprecated/folder is old classes ). I'm not sure this one was generated with the plugin i just moved and eventually renamed those old classes.

All new classes are generated with IDEA plugin.

File: src/sdk/model/edge/edge.ts (hunk 1 above)
Comment (cfsnyder) [reply src/sdk/model/edge/edge.ts]: still wondering about this @fradinni ?

**Comment by fradinni** [comment]
@cfsnyder updated

**Comment by cfsnyder** [comment]
thanks @fradinni

**Pull request merged**